### ISSUE TITLE
Unable to remove alias from schema

### ISSUE THREAD

**Issue by mgdigital** [edgedb/edgedb#4292]
- EdgeDB Version: 2.1
- OS Version: macOS Monterey

Steps to Reproduce:

1. Add an alias to a schema and create a migration
2. Remove the alias and create a migration:

```
did you drop alias 'default::MyAlias'? [y,n,l,c,b,s,q,?]
> y
edgedb error: EdgeDB could not resolve migration with the provided answers. Please retry with different answers.
```

Schema:
```
module default {

    type Foo {
        required property a -> str;
    }

    type Bar {
        required property b -> str;
    }

    alias MyAlias := (
      FOR _foo IN {(SELECT Foo)}
      UNION (
        (FOR _bar IN {(SELECT Bar)}
        UNION (
          (SELECT {
            single a := _foo.a,
            single b := _bar.b
          })
        ))
      )
    );
}

```

**Comment by msullivan** [comment]
A simplified version has just
```
    alias Foo := (for x in 1 union { x := 10 });
```

**Comment by aljazerzen** [comment]
This was (probably) closed by #4777
